src/context.rs

```diff
@@ -255,22 +255,29 @@ impl<'a> Context<'a> {
     }
 
     /// Will lazily get repo root and branch when a module requests it.
-    pub fn get_repo(&self) -> Result<&Repo, git2::Error> {
-        self.repo.get_or_try_init(|| -> Result<Repo, git2::Error> {
-            let repository = if env::var("GIT_DIR").is_ok() {
-                Repository::open_from_env()
-            } else {
-                let dirs: [PathBuf; 0] = [];
-                Repository::open_ext(&self.current_dir, git2::RepositoryOpenFlags::FROM_ENV, dirs)
-            }?;
-            Ok(Repo {
-                branch: get_current_branch(&repository),
-                workdir: repository.workdir().map(Path::to_path_buf),
-                path: Path::to_path_buf(repository.path()),
-                state: repository.state(),
-                remote: get_remote_repository_info(&repository),
+    pub fn get_repo(&self) -> Result<&Repo, git_repository::discover::Error> {
+        self.repo
+            .get_or_try_init(|| -> Result<Repo, git_repository::discover::Error> {
+                let shared_repo =
+                    ThreadSafeRepository::discover_with_environment_overrides(&self.current_dir)?;
+
+                let repository = shared_repo.to_thread_local();
+                let branch = get_current_branch(&repository);
+                let remote = get_remote_repository_info(&repository, branch.as_deref());
+                let path = repository.path().to_path_buf();
+                let trust_level = git_sec::Trust::from_path_ownership(&path)
+                    .unwrap_or(git_sec::Trust::Reduced);
+                Ok(Repo {
+                    repo: shared_repo,
+                    branch,
+                    // gitoxide might relativize the path, so we need to expand it again
```

Comment:
Is this not fixed in `gitoxide` by now? I thought it won't over-eagerly make paths relative anymore.

Reply:
Thanks, I hadn't updated that yet.